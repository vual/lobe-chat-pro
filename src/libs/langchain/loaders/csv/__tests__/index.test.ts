// @vitest-environment node
import * as fs from 'node:fs';
import { join } from 'node:path';
import { expect } from 'vitest';

import { CsVLoader } from '../index';

describe('CSVLoader', () => {
  it('should run', async () => {
    const content = fs.readFileSync(join(__dirname, `./demo.csv`), 'utf-8');

    const fileBlob = new Blob([Buffer.from(content)]);

    const data = await CsVLoader(fileBlob);

    expect(data).toMatchSnapshot();
  });
});
