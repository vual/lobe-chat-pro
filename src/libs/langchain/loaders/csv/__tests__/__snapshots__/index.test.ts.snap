// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CSVLoader > should run 1`] = `
[
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 1,
      "source": "blob",
    },
    "pageContent": ": 1
Hair: Black
Eye: Brown
Sex: Male
Freq: 32",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 2,
      "source": "blob",
    },
    "pageContent": ": 2
Hair: Brown
Eye: Brown
Sex: Male
Freq: 53",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 3,
      "source": "blob",
    },
    "pageContent": ": 3
Hair: Red
Eye: Brown
Sex: Male
Freq: 10",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 4,
      "source": "blob",
    },
    "pageContent": ": 4
Hair: Blond
Eye: Brown
Sex: Male
Freq: 3",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 5,
      "source": "blob",
    },
    "pageContent": ": 5
Hair: Black
Eye: Blue
Sex: Male
Freq: 11",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 6,
      "source": "blob",
    },
    "pageContent": ": 6
Hair: Brown
Eye: Blue
Sex: Male
Freq: 50",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 7,
      "source": "blob",
    },
    "pageContent": ": 7
Hair: Red
Eye: Blue
Sex: Male
Freq: 10",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 8,
      "source": "blob",
    },
    "pageContent": ": 8
Hair: Blond
Eye: Blue
Sex: Male
Freq: 30",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 9,
      "source": "blob",
    },
    "pageContent": ": 9
Hair: Black
Eye: Hazel
Sex: Male
Freq: 10",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 10,
      "source": "blob",
    },
    "pageContent": ": 10
Hair: Brown
Eye: Hazel
Sex: Male
Freq: 25",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 11,
      "source": "blob",
    },
    "pageContent": ": 11
Hair: Red
Eye: Hazel
Sex: Male
Freq: 7",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 12,
      "source": "blob",
    },
    "pageContent": ": 12
Hair: Blond
Eye: Hazel
Sex: Male
Freq: 5",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 13,
      "source": "blob",
    },
    "pageContent": ": 13
Hair: Black
Eye: Green
Sex: Male
Freq: 3",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 14,
      "source": "blob",
    },
    "pageContent": ": 14
Hair: Brown
Eye: Green
Sex: Male
Freq: 15",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 15,
      "source": "blob",
    },
    "pageContent": ": 15
Hair: Red
Eye: Green
Sex: Male
Freq: 7",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 16,
      "source": "blob",
    },
    "pageContent": ": 16
Hair: Blond
Eye: Green
Sex: Male
Freq: 8",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 17,
      "source": "blob",
    },
    "pageContent": ": 17
Hair: Black
Eye: Brown
Sex: Female
Freq: 36",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 18,
      "source": "blob",
    },
    "pageContent": ": 18
Hair: Brown
Eye: Brown
Sex: Female
Freq: 66",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 19,
      "source": "blob",
    },
    "pageContent": ": 19
Hair: Red
Eye: Brown
Sex: Female
Freq: 16",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 20,
      "source": "blob",
    },
    "pageContent": ": 20
Hair: Blond
Eye: Brown
Sex: Female
Freq: 4",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 21,
      "source": "blob",
    },
    "pageContent": ": 21
Hair: Black
Eye: Blue
Sex: Female
Freq: 9",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 22,
      "source": "blob",
    },
    "pageContent": ": 22
Hair: Brown
Eye: Blue
Sex: Female
Freq: 34",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 23,
      "source": "blob",
    },
    "pageContent": ": 23
Hair: Red
Eye: Blue
Sex: Female
Freq: 7",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 24,
      "source": "blob",
    },
    "pageContent": ": 24
Hair: Blond
Eye: Blue
Sex: Female
Freq: 64",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 25,
      "source": "blob",
    },
    "pageContent": ": 25
Hair: Black
Eye: Hazel
Sex: Female
Freq: 5",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 26,
      "source": "blob",
    },
    "pageContent": ": 26
Hair: Brown
Eye: Hazel
Sex: Female
Freq: 29",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 27,
      "source": "blob",
    },
    "pageContent": ": 27
Hair: Red
Eye: Hazel
Sex: Female
Freq: 7",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 28,
      "source": "blob",
    },
    "pageContent": ": 28
Hair: Blond
Eye: Hazel
Sex: Female
Freq: 5",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 29,
      "source": "blob",
    },
    "pageContent": ": 29
Hair: Black
Eye: Green
Sex: Female
Freq: 2",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 30,
      "source": "blob",
    },
    "pageContent": ": 30
Hair: Brown
Eye: Green
Sex: Female
Freq: 14",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 31,
      "source": "blob",
    },
    "pageContent": ": 31
Hair: Red
Eye: Green
Sex: Female
Freq: 7",
  },
  Document {
    "id": undefined,
    "metadata": {
      "blobType": "",
      "line": 32,
      "source": "blob",
    },
    "pageContent": ": 32
Hair: Blond
Eye: Green
Sex: Female
Freq: 8",
  },
]
`;
